di2f/index.md

```diff
@@ -0,0 +1,226 @@
+
+# Background
+
+## Decentralized Identifiers
+
+A decetralized identifier, or DID, is a string identifier of a _subject_, controlled by a _controller_. A DID might be used to encode a reference to an Ethereum address [@ethrDidResolver] on the Ethereum [buterin2014next,@wood2014ethereum] Mainnet, identify a resource on a network of IoT (Internet of Things) devices, or even represent a unique "identity" such as a user or organization [@ceramic2021Protocol]. For DIDs to be useful, they must be "resolvable" without reliance on a centralized network component. [@Fig:did-overview] shows the interactions between DID components.
+
+![The basic components of DID architecture. Source: [@reed2020decentralized] sec 1.3](./assets/did-overview.png){#fig:did-overview}
+
+The _Verifiable Data Registry_ in @fig:did-overview could be a blockchain or peer-to-peer network. The meaning of _verification_ differs between DID implementations -- some provide a high level of verification through blockchain transactions, while others rely purely on the assumption that the majority of peers are _good actors,_ e.g., IPID (an IPNS-based DID) or Ceramic.
+
+<!-- @sanderpick is this true re: Ceramic? I'm not sure I follow this statement fully? -->
+
+A DID can identify any actor or structure in the network (the _subject_), and should resolve to a _document_ that is accessible from anywhere in the network. Formally, a DID document describes the _subject_, and the document is controlled by one or more _controllers_. In short, DIDs point to relationships between the components of a decentralized network. What can this component do? Who/what can do it? Who/what dictates who can do it? See [@reed2020decentralized] for the DID specification.
+
+## Threads
+
+A thread is topic-based collection of single-writer logs. A collection of logs represent updates to the "state" of an object (or dataset). The basic units of a thread -- logs and records -- provide a framework for creating, storing, and transmiting data in a p2p distributed network. The thread protocol is outline in detail in the threads whitepaper [@pick2020protocol].
+
+A log within a thread is essentially a set of cryptographically linked (i.e., hash-linked) records, that form a specific type of Merkle-DAG (directed acyclic graph) that represents a purely functional [@okasaki1999purely] and authenticated [@tamassia2003authenticated] (i.e., immutable) singly-linked list. The key insight here is that, assuming two peers have received all of the same updates to a thread, they will deterministically arrive at the same thread structure, and that thread structure can be summarized by the (set of) hash(es) of the head(s) of the underlying log(s).
+
+# Thread DIDs
+
+<!-- @sanderpick What are we trying to "solve" or say in this section specifically? Why, in one or two sentences, do we need to build this? -->
```

Comment:
@sanderpick What are we trying to "solve" or say in this section specifically? Why, in one or two sentences, do we need to build this?

Reply:
I don't think I have anything else to add to the motivation beyond the top-level bullets that were in the GH discussion... not sure I follow your question though

Reply:
Those points are now in the conclusions, but they are more about why "threads would benefit from a native DID". What I'd like to get at here is more" why should someone who's new to the space care about threads + dids. What is magic here? Something along the lines of having a way to uniquely identify an append-only log on the network, without any "centralized" coordination is important, because it allows developers to leverage access controlled storage APIs _without_ having to worry about API keys, without risk of losing the underlying data, with increased user-control of data, with... but I need help with this bit of the story.

Reply:
I like that... and then also the ability to advertise thread-based services to the network and allow crypto-native / web3 users the ability to leverage / pay for them without API keys?

Reply:
Niiice!